gadget/update.go

```diff
@@ -312,6 +312,21 @@ func RemodelUpdatePolicy(from, _ *LaidOutStructure) bool {
 	return true
 }
 
+// KernelUpdatePolicy implements the update policy for kernel asset updates
+func KernelUpdatePolicy(from, to *LaidOutStructure) bool {
+	for _, rn := range to.ResolvedContent {
+		if rn.KernelUpdateFlag {
+			// XXX: if there is a structure that has mixed
+			// content from $kernel: or gadget then this will
+			// also update the gadget data. So we will need
+			// a per ResolvedContent predicator?
```

Comment:
yes, it seems we need to support the lower level granularity as well. Given that now we have explicit  ResolvedContent structures, it should be easier, no?